engine/src/main/scala/cromwell/server/CromwellShutdown.scala

```diff
@@ -0,0 +1,222 @@
+package cromwell.server
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import akka.Done
+import akka.actor.{ActorRef, ActorSystem, CoordinatedShutdown, _}
+import akka.http.scaladsl.Http
+import akka.pattern.{AskTimeoutException, GracefulStopSupport}
+import akka.routing.Broadcast
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import cats.instances.future._
+import cats.syntax.functor._
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.workflow.WorkflowManagerActor.{AbortAllWorkflowsCommand, PreventNewWorkflowsFromStarting}
+import cromwell.services.SingletonServicesStore
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+/**
+  * Collection of methods and objects used to control Cromwell graceful shutdown process.
+  */
+object CromwellShutdown extends GracefulStopSupport {
+  private val logger = LoggerFactory.getLogger("CromwellShutdown")
+  
+  // Includes DB writing actors, I/O Actor and DockerHashActor
+  private val PhaseStopIoActivity = "stop-io-activity"
+  // Shutdown phase allocated when "abort-jobs-on-terminate" is true to give time to the system to abort all workflows
+  // This phase is at the same level as PhaseServiceRequestsDone in the dependency graph.
+  private val PhaseAbortAllWorkflows = "abort-all-workflows"
+
+  /**
+    * Configuration overriding pre-existing phases to wire in custom phases in the dependency graph.
+    */
+  val OverrideShutdownConfiguration = ConfigFactory.parseString(
```

Comment:
See my answer to your other comment but in short yes as long as we're ok with making it mandatory for a user to include `application.conf` in their conf.